src/main/java/edu/byu/cs/service/AdminService.java

```diff
@@ -0,0 +1,153 @@
+package edu.byu.cs.service;
+
+import edu.byu.cs.analytics.CommitAnalyticsRouter;
+import edu.byu.cs.canvas.CanvasException;
+import edu.byu.cs.canvas.CanvasService;
+import edu.byu.cs.canvas.model.CanvasSection;
+import edu.byu.cs.dataAccess.DaoService;
+import edu.byu.cs.dataAccess.DataAccessException;
+import edu.byu.cs.dataAccess.ItemNotFoundException;
+import edu.byu.cs.dataAccess.UserDao;
+import edu.byu.cs.honorChecker.HonorCheckerCompiler;
+import edu.byu.cs.model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+
+public class AdminService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminService.class);
+
+    public static Collection<User> getUsers() throws DataAccessException {
+        UserDao userDao = DaoService.getUserDao();
+
+        Collection<User> users;
+        try {
+            users = userDao.getUsers();
+        } catch (DataAccessException e) {
+            LOGGER.error("Error getting users", e);
+            throw e;
+        }
+
+        return users;
+    }
+
+    public static void updateUser(User user) throws DataAccessException, ItemNotFoundException {
+        UserDao userDao = DaoService.getUserDao();
+        User existingUser;
+        try {
+            existingUser = userDao.getUser(user.netId());
+        } catch (DataAccessException e) {
+            LOGGER.error("Error getting user", e);
+            throw e;
```

Comment:
Agreed, but not in this PR--I'm working on an overhaul of all the Handler/Service exception handling. This PR is focused on extracting the Service layer while preserving logical structure.

Reply:
I'm beginning to detect that we have different attitudes about the role of a PR. Again, these are kinds of discussions are stylistic/conventional measures; I don't want to say either of us are right, but it will help us collaborate better together if we can assimilate and be in the same page.

It seems like there are two different perspectives that can be taken towards a PR:
* A PR is like a commit: one comes now and another will change it later
* A PR is like a branch: many commits will be added to each other until the meaningful milestone or objective is complete

**TL;DR** PRs serve as an organizational unit of changes that achieve a business goal. Use commits to express steps of the work, but keep related changes contained within the same PR.


I personally believe a PR should be more like a branch-- it can contain many commits, potentially including many that individually touch lots of files, and in the end, the PR will correlate closely to an actual business result. It may be that I discover halfway through a PR, that I receive new information which invalidates some of the assumptions made in the code, and so I would either build on the PR to fix it, or I may restart with a new PR to try a new approach.

The aligns with my experience where certain business needs are the end goal, and multiple people collaborate together on a PR to make sure that the result is the best possible for the company.

In my experience, the purpose of a PR is to represent something to the business (not typically a mere minor technical improvement). A PR, or a a small handful of PRs, will be what I show my boss to show him what I did.

In certain cases, for the sake of urgency or sheer size, issues should be split into smaller items which would result in parallel reductions in the overall size of a corresponding PR. However, there is no large requirement here and no urgency to merge in the portion of code completed before we totally finish it.

The requested changes can be represented in a commit or two, and I think it would be totally appropriate to put them together under the banner of "Refactoring and Extracting Controller Code into the proper Locations." Breaking down the work into meaningful chucks at the commit level makes sense, but that level of detail should not directly equate to a linear number of PRs as well.

Reply:
All right, what do you think is the best way to do that here then? I could introduce an intermediate refactoring to address just the requested changes, but it'll get completely replaced in the second PR anyway; or I could compile that entire PR into this one, but that will be a lot more than just a couple of commits--it would involve adding several new Exception classes and things of that ilk.

Reply:
@frozenfrank I'm not sure a PR needs to be so broad though. Extracting service logic from controllers does seem like a big change, and other overhauls might be good to do separately.

Reply:
@webecke Fair point. Let's wrap up this one, and represent related changes in other issues.